operations/ftdc.go

```diff
@@ -737,19 +737,82 @@ func toT2() cli.Command {
 			ctx, cancel := context.WithCancel(context.Background())
 			defer cancel()
 
-			inputFile, err := os.Open(inputPath)
+			inputStat, err := os.Stat(inputPath)
 			if err != nil {
 				return errors.Wrapf(err, "problem opening file '%s'", inputPath)
 			}
-			defer func() { grip.Warning(inputFile.Close()) }()
 
-			// All genny output files are named using the workload actor and operation.
-			//   i.e., Actor.Operation.ftdc
-			//
-			// We get the actor and operation names from the ftdc filepath for use
-			// in the translation process.
-			fileName := filepath.Base(inputPath)
-			actorOperation := strings.Split(fileName, ".ftdc")
+			// Stores ftdc chunk iterators and its metadata to be used in TranslateGenny.
+			var outputSlice []*ftdc.GennyOutputMetadata
+
+			// Check if input path is an individual file or directory.
+			// All ftdc files in the provided input directory are initially
+			// used to set GennyOutputMetadata Name, Iter, StartTime and EndTime.
+			switch mode := inputStat.Mode(); {
+			// Handle directory.
+			case mode.IsDir():
+				if !strings.HasSuffix(inputPath, "/") {
+					inputPath += "/"
+				}
+				input, err := os.Open(inputPath)
+				if err != nil {
+					return errors.Wrapf(err, "problem opening input path '%s'", inputPath)
+				}
+				files, err := input.Readdir(-1)
+				if err != nil {
+					return errors.Wrapf(err, "problem reading dir '%s'", inputPath)
+				}
+
+				for _, file := range files {
+					if file.Mode().IsRegular() && filepath.Ext(file.Name()) == ".ftdc" {
+						var gennyOutput ftdc.GennyOutputMetadata
+
+						f, err := os.Open(inputPath + file.Name())
+						if err != nil {
+							return errors.Wrapf(err, "problem opening file '%s'", inputPath+file.Name())
+						}
+						defer func() { grip.Warning(f.Close()) }()
+						gennyOutput.Iter = ftdc.ReadChunks(ctx, f)
+						gennyOutput = ftdc.GetGennyTime(ctx, gennyOutput)
+
+						// Reset the chunk iterator for the ftdc file.
+						input.Close()
+						f, err = os.Open(inputPath + file.Name())
+						if err != nil {
+							return errors.Wrapf(err, "problem opening file '%s'", inputPath+file.Name())
+						}
+
+						gennyOutput.Iter = ftdc.ReadChunks(ctx, f)
+						gennyOutput.Name = strings.Split(file.Name(), ".ftdc")[0]
+						// Upon describing the genny output metadata, store into the slice
+						// to be used in TranslateGenny.
+						outputSlice = append(outputSlice, &gennyOutput)
+					}
+				}
+			// Handle single file.
+			case mode.IsRegular():
+				var gennyOutput ftdc.GennyOutputMetadata
+				input, err := os.Open(inputPath)
+				if err != nil {
+					return errors.Wrapf(err, "problem opening file '%s'", inputPath)
+				}
+				defer func() { grip.Warning((input.Close())) }()
+
+				gennyOutput.Iter = ftdc.ReadChunks(ctx, input)
+				gennyOutput = ftdc.GetGennyTime(ctx, gennyOutput)
+				input.Close()
+
+				input, err = os.Open(inputPath)
```

Comment:
Do we need to open this file twice?

Reply:
I open it the second time to get a new chunk iterator that starts at the beginning of the file.